app/src/main/java/org/flyve/mdm/agent/core/mqtt/MqttModel.java

```diff
@@ -192,12 +192,13 @@ public void connect(final Context context, final MqttCallback callback) {
             options.setUserName(mUser);
             options.setCleanSession(true);
             options.setMqttVersion(MqttConnectOptions.MQTT_VERSION_3_1);
-            options.setConnectionTimeout(0);
+            options.setConnectionTimeout(MqttConnectOptions.CONNECTION_TIMEOUT_DEFAULT);
+            options.setKeepAliveInterval(MqttConnectOptions.KEEP_ALIVE_INTERVAL_DEFAULT);
             options.setAutomaticReconnect(true);
 
             // Create a testament to send when MQTT connection is down
-            String will = "{ online: false }";
-            options.setWill("/Status/Online", will.getBytes(), 0, false);
+            String will = "{ \"online\": false }";
+            options.setWill(mTopic + "/Status/Online", will.getBytes(), 0, true);
```

Comment:
the online status belongs to the device. /status/Online is a subtopic of the device.